### DISCUSSION TITLE
PrimeFlex 4.0 is Coming Soon!🎉

### DISCUSSION THREAD

**Discussion by atkntepe** [primefaces/community#22] [PrimeFlex] [answered]
PrimeFlex 4.0 is Coming Soon!🎉

Hey everyone, we're thrilled to share that PrimeFlex 4.0 is on its way! Here's what you can look forward to:

🛠 Core Features:

- PostCSS Support
- Customizations

🎨 New Aesthetic Features:

- Filters & Backdrop Filters (blur, brightness, grayscale, etc.)
- RGBA Colors
- Linear & Radial Gradient Classes
- Font Families
- Shadow colors
- SVG Classes
- Outline styles

📏 Layout & Sizing:

- New width and min|max-width
- New height and min|max-height

🔥 And much more!

We're targeting to release this month(Sep.), so stay tuned for more updates.

**Comment by jepsar** [comment]
Is there a migration guide? If so I think we will need to extend the [CLI tools](https://github.com/primefaces/primefaces/tree/master/primefaces-cli) with a 3 to 4 migration function.

**Reply by atkntepe** [reply]
At the moment, we are finalizing new features and also writing the new docs. End of this week, everything will be clear. After that, we will work on migration

**Reply by melloware** [reply]
@atakantepe are you taking over the VSCode Snippets too?  If so @FlipWarthog has a tool to autogenerate the snippets using Rust code to read the whole CSS and generate all the proper snippets.

See: https://github.com/primefaces/primeflex/issues/152

See: https://github.com/FlipWarthog/css-to-snippets

**Reply by atkntepe** [reply]
Yes, we will also update the VsCode Snippets after the release of 4.0. I will definitely check out what @FlipWarthog made. Thanks

**Comment by csakis** [comment]
what happened? This post is 6 months old and no news about PrimeFlex

**Comment by Picoquio** [comment]
Thank you for all the work you guys put into primeflex! Are there any news regarding the release date? It's been a while

**Comment by cagataycivici** [comment]
We have decided to sunset the project due to overlap of upcoming functionality. New replacements are layout, typography, spacing components within core libraries and also a migration tool that will statically update PrimeFlex code to Tailwind or to new Prime components e.g. Box, Grid, Text.

So in summary, majority of the PrimeFlex features will move into core as new UI components.

**Reply by csakis** [reply]
That sounds great. I am really looking forward to it.

**Reply by haysmike** [reply]
Hi @cagataycivici ! Sorry for the uninformed question, but what does "core" refer to here?

**Reply by jumika** [reply]
I guess primevue and primereact.

**Reply by ugumerie** [reply]
What migration tool can we use to migrate to tailwind?

**Reply by stylepatrick** [reply]
Is there already a migration tool? I only know the one mentioned here:
https://primeng.org/guides/primeflex

Unfortunately we can't migrate Grid and Cols, which are some core components. Is there a migration tool planned to enable a smooth migration from Primeflex to TailwindCSS? 

As far as I know so far, it takes a lot of effort to update all projects.

**Comment by csakis** [comment]
>  "New replacements are layout, typography, spacing components within core libraries..."

I am looking at the Primevue documentation site and find no utility classes mentioned there. It's been almost a year since the last message. Can we get an update on what happened with the PrimeFlex integration?
I assume we should just use Tailwind.

**Comment by skj-skj** [comment]
just noticed primeflex is bumped to 4.0.0 on [npm](https://www.npmjs.com/package/primeflex/v/4.0.0)

**Reply by aheritier** [reply]
@atakantepe is it an official release before closing the project?
Sources here are not updated for 2 years....

**Reply by aheritier** [reply]
@cagataycivici it seems you created a 4.0.0 branch but you didn't merged it into master ?

**Reply by boris-bojic-pid** [reply]
I am wondering, too ... maybe it's a "last" release to publish all the work that was done?

**Comment by stylepatrick** [comment]
I am now puzzled. Officially PrimeNg said that we should switch to TailwindCSS (https://primeng.org/tailwind), so why is Primeflex still maintained and updated. What should we use now Primeflex or TailwindCSS? Is there anything official from PrimeNg?

**Comment by sephentos** [comment]
@atakantepe 

Could you give us an update regarding this please? 4.0.0 released on npm and no infos?

**Reply by stylepatrick** [reply]
I just found following info at the bottom of the article:

https://primeng.org/guides/migration

**Comment by cagataycivici** [comment, marked as answer]
Plans have changed since Sep 13, 2023, PrimeFlex is sunsetting for sure. v4 is released to bring compatibility with new design token based theming only. Other than that, it is same as v3.
